Emby server (depends on https://github.com/SynoCommunity/spksrc/pull/1662)

cytec opened the pull request:
Latest emby server...
This depends on `ffmpeg>=2.6-3` `imagemagick=6.9.0-1` and `mediainfo=0.7.72-1`

basically for this to work, those PR's must be merged/published first:

ffmpeg: https://github.com/SynoCommunity/spksrc/pull/1662
ImageMagick: https://github.com/SynoCommunity/spksrc/pull/1672

Diaoul commented on spk/emby/Makefile:
```diff
@@ -0,0 +1,52 @@
+# Notify of unsupported architectures
+ifneq ($(findstring $(ARCH),bromolow cedarview evansport x86 armada370 88f6281 armadaxp comcerto2k avoton x64 armada375 alpine),$(ARCH))
+$(error Sorry $(ARCH) is not a supported architecture)
+endif
```
@Dr-Bean: you did add some UNSUPPORTED_ARCH variable in recent commits, right?

Dr-Bean replied:
Yup, here: https://github.com/SynoCommunity/spksrc/commit/53411f4424e8b491e8d7f1cc9c84045615d038b5.

Dr-Bean commented on spk/emby/Makefile:
```diff
@@ -0,0 +1,52 @@
+ADMIN_PORT = 8096
+RELOAD_UI = yes
+DISPLAY_NAME = Emby
+CHANGELOG = "1. Update emby to 3.0.5597.1<br>2. depend on ffmpeg >= 2.6-3"
```
You're not actually depending on ffmpeg `>=` 2.6-3, according to L11.
I'd recommend using the same approach (adding a `>`, I mean) for imagemagick and mediainfo, otherwise updating those packages would cause issues.

cytec replied:
yeah that shouldn't be a problem for `mediainfo` and `ffmpeg` but `imagemagick` is linked to `libMagickWand-6.Q16.so` so updating `imagemagick` __can__ cause issues indeed

Dr-Bean replied:
Yes, but imo that means you need to update both packages at the same time when major upgrades to ImageMagick occurs, which is understandable. 
I was thinking more along the lines of updating ImageMagick to `imagemagick 6.9.0-2`, or a minor update to say 6.9.1 which doesn't necessarily mean the patch needs changing.

cytec replied:
now i got it :)
will change that

Dr-Bean commented on cross/emby/patches/imagick.patch:
```diff
@@ -0,0 +1,10 @@
+   <dllmap dll="CORE_RL_Wand_.dll" target="libMagickWand-6.so" os="freebsd,openbsd,netbsd"/>
+   <dllmap dll="CORE_RL_Wand_.dll" target="./MediaInfo/osx/libmediainfo.dylib" os="osx"/>
+ </configuration>
+\ Kein Zeilenumbruch am Dateiende.
```
I don't think this last line should be included, right?

cytec replied:
oops, thats right

Dr-Bean commented on spk/emby/Makefile:
```diff
@@ -0,0 +1,52 @@
+
+.PHONY: emby_extra_install
+emby_extra_install:
+	install -m 755 -d $(STAGING_DIR)/
```
This seems unnecessary, no?

Dr-Bean commented on spk/emby/src/dsm-control.sh:
```diff
@@ -0,0 +1,86 @@
+INSTALL_DIR="/usr/local/${PACKAGE}"
+PATH="${INSTALL_DIR}/bin:${PATH}"
+USER="${PACKAGE}"
+USER_HOME="$(eval echo ~$USER)"
```
Maybe just set it to the same path you use in the installer (https://github.com/SynoCommunity/spksrc/pull/1673/files#diff-af6eb7962b714d2e23e6d6895ba91cd0R38). I see you also define `DATA_DIR`, it's the same thing really.

Dr-Bean replied:
Just noticed: is `$(HOME_DIR}` used anywhere at all, or am I missing it?

cytec replied:
i think this isn't even used at all, just a leftover from the nzbdrone installer...
i _think_ that can be deleted

Dr-Bean commented on cross/emby/Makefile:
```diff
@@ -0,0 +1,27 @@
+	tar -cf - -C $(WORK_DIR)/$(PKG_DIR) . | tar -xf - -C $(STAGING_INSTALL_PREFIX)/share/$(PKG_DIR)
+
+myExtract:
+	unzip ../../distrib/$(PKG_DIST_NAME) -d $(WORK_DIR)/$(PKG_DIR)
```
If the reason for this is an incorrect `EXTRACT_PATH`, then may I suggest to specify `EXTRACT_PATH = $(WORK_DIR)/$(PKG_DIR)` instead of the current `EXTRACT_TARGET` override?
 
See e.g. [selfoss](https://github.com/SynoCommunity/spksrc/blob/03f583a3db38b80021aa9e7c5af641ac2b9fdf0b/cross/selfoss/Makefile#L14)

cytec replied:
honestly i don't remember from memory why i did this...
I'll check and if its because of the `EXTRACT_PATH` change it...

i think there was something like a folder inside the `zip` file which i'd like to exclude or something like that

Dr-Bean commented on spk/emby/Makefile:
```diff
@@ -0,0 +1,49 @@
+SPK_NAME = emby
+SPK_VERS = 3.0.5597.1
+SPK_REV = 3
```
`1` here?

cytec replied:
leftover from testing, will fix that

Dr-Bean commented on spk/emby/src/dsm-control.sh:
```diff
@@ -0,0 +1,85 @@
+EXTRA_LIBS="/usr/local/mediainfo/lib:/usr/local/imagemagick/lib:${INSTALL_DIR}/lib"
+FFMPEG="/usr/local/ffmpeg/bin"
+
+DATA_DIR="${INSTALL_DIR}/var"
```
Personally, I'm not a fan of adding this type of variable, where it's not much of a shortcut at all. Simply referring to `{INSTALL_DIR}/var` where you need it works fine. (PID_FILE and `-programdata` in COMMAND)

cytec replied:
you are right!

Dr-Bean commented on spk/emby/src/installer.sh:
```diff
@@ -0,0 +1,99 @@
+INSTALL_DIR="/usr/local/${PACKAGE}"
+TMP_DIR="${SYNOPKG_PKGDEST}/../../@tmp"
+INSTALL_LOG="${INSTALL_DIR}/var/install.log"
+TMP_INSTALL_LOG="${TMP_DIR}/${PACKAGE}/var/install.log"
```
Leftover from somewhere, it seems :)

Dr-Bean commented on spk/emby/src/installer.sh:
```diff
@@ -0,0 +1,99 @@
+    ln -s ${SYNOPKG_PKGDEST} ${INSTALL_DIR}
+
+    # Logging Install
+    mkdir -p ${INSTALL_DIR}/var
```
The dir is already created at this point, right? You specified it in `emby_extra_install`

cytec replied:
yup, leftover

Dr-Bean commented on spk/emby/src/installer.sh:
```diff
@@ -0,0 +1,99 @@
+preupgrade ()
+{
+    # Log Upgrade
+    echo "|| Beginning Package Upgrade - $(date) ||" >> ${INSTALL_LOG}
```
You might have reused Sonarr's code? I know that logging (initially) was interesting for Sonarr, but does Emby have the same level of 'possible issues' that an upgrade should be logged like this? It's not all that likely we'll be doing anything with the info, because it's not all that informative tbh ;)
Fwiw, I'm fine with leaving it in during the beta, but outside of that, I'd propose to remove it.

cytec commented:
@Dr-Bean i've changed some stuff (you mentioned) and updated emby to the most recent version... 
something seemed to go wrong when i run rebase (did a lot of extra commits i wasn't expecting, but besides from that "beauty" stuff, everything worked well) maybe you'll have a look again so we are finally able to get this out in the wild?

Dr-Bean commented on cross/emby/Makefile:
```diff
@@ -0,0 +1,24 @@
+PKG_EXT = zip
+PKG_DIST_NAME = MediaBrowser-$(PKG_VERS).$(PKG_EXT)
+PKG_DIST_SITE = https://github.com/MediaBrowser/MediaBrowser.Releases/raw/master/Server
+PKG_DIR = emby
```
`$(PKG_NAME)`. 

Btw, was MediaBrowser the old name or something? Otherwise I'd assume you use that for PKG_NAME etc.

cytec replied:
yup... MediaBrowser was renamed to emby

Dr-Bean commented on cross/emby/Makefile:
```diff
@@ -0,0 +1,24 @@
+
+HOMEPAGE = http://emby.media/
+COMMENT  = Emby streams everything Your personal videos, music, photos, and live television in the palm of your hand.
+LICENSE  =
```
No license?

Dr-Bean commented on spk/emby/Makefile:
```diff
@@ -0,0 +1,49 @@
+SPK_NAME = emby
+SPK_VERS = 3.0.5724.5
+SPK_REV = 5
```
Let's start at 1.

Dr-Bean commented on spk/emby/Makefile:
```diff
@@ -0,0 +1,49 @@
+CHANGELOG = "Update emby to 3.0.5724.5"
+
+HOMEPAGE = http://emby.media/
+LICENSE  =
```
License please ;)

Dr-Bean commented on spk/emby/src/dsm-control.sh:
```diff
@@ -0,0 +1,82 @@
+
+start_daemon ()
+{
+    start-stop-daemon -c ${USER} -S -q -b -m -p ${INSTALL_DIR}/var/emby.pid -N 10 -x ${COMMAND} > /dev/null
```
Fwiw, I wouldn't mind if you set `PID_FILE = ${INSTALL_DIR}/var/emby.pid` on L15 or thereabouts, and did ${PID_FILE} here. That's the usual approach.

Dr-Bean commented on spk/emby/src/installer.sh:
```diff
@@ -0,0 +1,91 @@
+    # Add firewall config
+    ${SERVICETOOL} --install-configure-file --package ${FWPORTS} >> /dev/null
+
+    # Log
```
I can't remember, but wasn't this supposed to be removed? Or did you want to keep it around for the beta?

Dr-Bean commented on spk/emby/src/installer.sh:
```diff
@@ -0,0 +1,91 @@
+    mv ${TMP_DIR}/${PACKAGE}/var ${INSTALL_DIR}/
+
+    exit 0
+}
```
Fix your editor already :P

Dr-Bean commented:
Just looked, didn't test, but other than that, just a couple comments, nothing too interesting. 

I skipped the review of my own commits ;) Do you want to retry that rebase? A `git rebase master` should usually do, I think.

cytec commented:
I'll fix those comments and try another rebase when i get home tonight

LukePulverenti commented:
Is this able to be merged in?

G1zm0 commented:
ImageMagick is no longer a dependency because the devs use embymagick from 3.0.5781.6 on.

"Beta and Dev releases are now using the package embymagick for ImageMagick libraries. embymagick does not get installed as a system library, so there should no longer be conflicts with distribution releases of ImageMagick."

I'm hoping you'll have some time on your hand this period. :-)

cytec commented:
@LukePulverenti so i/we need to update this to include embymagick instead of imagemagick? or will it work if imagemagick is installed?

LukePulverenti commented:
No, this isn't true. ImageMagick is still a dependency. We have created embyimagemagick on the open build service for our packages there as an alternative to avoid conflicts. All other packages should still use ImageMagick.

LukePulverenti commented:
However - good news - MediaInfo is no longer used and can be eliminated.

fonix232 commented:
@cytec Sorry for pingig you here, didn't know any better.

Could you please add the emby and ffmpeg builds you're currently hosting as beta packages to your SPK repo? For some reason I am not able to install any packages manually (I just get an "Operation failed").

cytec commented:
@fonix232 sure you got the correct arch?
idk i have to take a look maybe i can add them manually to the database, if so, i'll do it

fonix232 commented:
Well given that I'm running Xpenology (sorry), it's a bromolow x86_64 base - same support as defined in the x64 package (it does define my branch, bromolow, so I suppose it should be working). Funny thing is, I have the same if I try to manually install packages - but if the same package is added from a repo, it installs fine.

cytec commented:
@fonix232 i don't know if this has any impact on manual install but i don't have those SPK's signed so you'll may try to enable installing SPK's from anyone instead of trusted publishers in DSM

fonix232 commented:
I haver that too - I use packages from other sources so I needed it.

LukePulverenti commented:
cytek - is there anything we (Emby) can do to help expedite this and get Emby listed in the Synology catalog? Can we take on some of the tasks remaining ourselves? We would really like to get this done. Thanks.

cytec commented:
@LukePulverenti nope, sadly not. AFAIK the emby package is good to go but isn't really useful without ffmpeg which still needs some cleanup...

LukePulverenti commented:
Cytec, thanks. Do you have a contact email? There is one thing I'd like to go over with you outside of here. Thanks.

cytec commented:
@LukePulverenti yup: iamcytec at googlemail dot com

cytec commented:
@LukePulverenti could you please tell me which file from https://github.com/MediaBrowser/MediaBrowser.Releases/raw/master/Server is the correct one to use?

LukePulverenti commented:
Correct. You can get Emby.Mono.zip from the server release tag:

https://github.com/MediaBrowser/Emby/releases/latest

Is that sufficient or do you need a permanent url that you can link to for automation purposes?

cytec commented:
nope should be good i think... gonna test that and report back

maximm07 commented:
Is it possible/planned to include HW transcoding support for Intel based models like DS216+ etc..tnx

rrig commented:
I have two Synology NAS boxes and would really like to see Emby in the catalog.

boba51 commented:
I would love to see Emby supported on my Synology NAS!

scyto commented:
Yeah a fully emby package on the syno community (or better yet in the synology repo) would be awesome!

techabyte commented:
YES! This would be awesome to have on the Syno repo. Seriously Plex lifetime sub is now $150 USD :/ having this option to try out would be nice.

GoodOmens83 commented:
Compiled this and ffmpeg and this is working wonderfully on my DS1515+ 

If only I had a DS716+ to test out ffmpeg/intel sync transcoding.

LukePulverenti commented:
Great stuff, thanks!

hamzalija commented:
Is there a chance to get a spk file for our synology boxes with a newer stable version any time soon, since i have had amany u pdating problems, and stabiliity issues regarding old version. 

updating it is causing the service to start but not accessable trougth the network and completely useless. 

the only solution to the "update" is to "reinstall" the old version that i have laying arround in spk file. ;( ;( 
and spend hours and hours redoing everything.

cytec commented:
im currently working on the ffmpeg package which kinda holds this back... once that is finished there should be a newer version ;)

horstepipe commented:
hey
can DS414 users (armadaxp) still hope that there will come a working version?

baskient commented:
Hi there,
just wanted to let you guys know that I love emby. I was using it before my nas (Synology DiskStation DS215j) updated to DSM 6 automatically last night. Now it stopped working.There is no FFMPEG package for my nas so I ended up editing the dependencies of the info file of the emby_armada375 package to remove the FFMPEG dependencies. I have been using it in that way in kodi for quite some time. So in theory FFMPEG is not mandatory if you are playing the files in kodi through a direct network link. So would it be possible for me to run this new package that should be working under DSM 6 without the FFMPEG dependency?

hex92 commented:
Hello,

I'm on DS414 too and having no luck with the special build mentionned here : http://emby.media/community/index.php?/topic/32246-dsm-60/?hl=5882#entry310426

I made the modification with sudo without success...
Emby try to launch and fail without a message (nothing in /var/log/messages or syslog)

hex92 commented:
cytec - if you need anyone to compile it for test, i will be glad to help :)

hex92 commented:
cytec - does the problem has to do more with mono specific package from synocommunity than ffmpeg ? I figured out to launch emby by resolving this problem (segfault) >> http://emby.media/community/index.php?/topic/32554-dsm-6-ds414-armadaxp/

cytec commented:
well your problem is more due to the fact that you use a platform for which currently aren't any toolchains available and AFAIK none of the SynoCommunity devs (we like to keep our devices stable too) has currently installed so ... i would say it's most likely a problem with the installed version of momo which seems to not work as expected... thought you'll maybe try the officiall synology mono package and then come back to this and just change the values/paths

hex92 commented:
cytec - that what's i have done two days ago :) and it's working perfectly !

BrianGilbert commented:
+1 on using Synology version of Mono.

solabc16 commented:
Evening - just dropping a note here to reference this thread over on the Emby site http://emby.media/community/index.php?/topic/32321-getting-the-coverart-plugin-working/?p=313724

- James

cytec commented:
@LukePulverenti i'm currently changing ImageMagick to `8Q` and was thinking about updating it to `7.0.1` as well... does emby play well/work with `7.0.1`? as it seems the default lib in the config is still 6 so i'll need to update the patch as well...

cytec commented:
@solabc16 @LukePulverenti seeing that there are plans for a own repo and publishing emby through the `Emby LLC` Repo (for reference see: http://emby.media/community/index.php?/topic/34866-synology-package-in-development/?p=331583) which means i can stop working on the emby Package and you'll provide your own Package/Repo? Or are you planing on just copy/pasting the code all together?

LukePulverenti commented:
I believe 7 has some breaking changes that we'll need to adjust for first.

LukePulverenti commented:
We still want to be published in the Synology app center. that is the top priority.

cytec commented:
ok, then i'll just update the package to the `6.9.4` and change the bit depth to 8 for now :)

jose-pr commented:
Now that fmpeg has been uodated any chance of pushing this package?

cytec commented:
I'll have a look at this when i get home and hopefully get it done soon

Dr-Bean commented:
Closing, following closure of #1346
